lib/Controller/SignalingController.php

```diff
@@ -196,6 +208,45 @@ public function getSettings(string $token = ''): DataResponse {
 		return new DataResponse($data);
 	}
 
+	private function getFederationSettings(?Room $room): array {
+		if ($room === null || !$room->isFederatedConversation()) {
+			return [];
+		}
+
+		try {
+			$participant = $this->participantService->getParticipant($room, $this->userId);
+		} catch (ParticipantNotFoundException $e) {
+			return [];
+		}
+
+		/** @var \OCA\Talk\Federation\Proxy\TalkV1\Controller\SignalingController $proxy */
+		$proxy = \OCP\Server::get(\OCA\Talk\Federation\Proxy\TalkV1\Controller\SignalingController::class);
+		$response = $proxy->getSettings($room, $participant);
+
+		if ($response->getStatus() === Http::STATUS_NOT_FOUND) {
+			return [];
+		}
+
+		/** @var TalkSignalingSettings $data */
+		$data = $response->getData();
+
+		return [
+			'server' => $data['server'],
+			'nextcloudServer' => $this->prependProtocolIfNotAvailable($room->getRemoteServer()),
```

Comment:
Should not be necessary anymore since the latest changes in server and talk

```suggestion
			'nextcloudServer' => $room->getRemoteServer(),
```